website/static/js/koHelpers.js

```diff
@@ -436,6 +436,48 @@ ko.bindingHandlers.datePicker = {
 };
 
  /**
+ * Bind content of contenteditable to observable. Looks for maxlength attr
+ * and underMaxLength binding to limit input.
+ * Example:
+ * <div contenteditable="true" data-bind="editableHTML: <observable_name>" maxlength="500"></div>
+ */
+ko.bindingHandlers.editableHTML = {
+    init: function(element, valueAccessor, allBindings, bindingContext) {
+        var $element = $(element);
+        var initialValue = ko.utils.unwrapObservable(valueAccessor());
+        $element.html(initialValue);
+        $element.on('change input paste keyup blur', function() {
+            var observable = valueAccessor();
+            observable($element.html());
+        });
+    },
+    update: function(element, valueAccessor, allBindings, bindingContext) {
+        var $element = $(element);
+        var initialValue = ko.utils.unwrapObservable(valueAccessor());
+        var bindings = bindingContext;
+        var charLimit = $element.attr('maxlength');
+        if (charLimit && bindingContext.underMaxLength) {
+            var inputTextLength = $element[0].innerText.length;
+            var errorExceedsCharacterLimit = 'Exceeds character limit. Please reduce to ' + charLimit + ' characters or less.';
+            if (inputTextLength > parseInt(charLimit) + 1) {
+                bindingContext.underMaxLength(false);
+                if (bindingContext.errorMessage) {
+                    bindingContext.errorMessage(errorExceedsCharacterLimit);
+                }
+            } else {
+                bindingContext.underMaxLength(true);
+                if (bindingContext.errorMessage) {
+                    bindingContext.errorMessage('');
+                }
+            }
+        }
+        if (initialValue === '') {
+            $(element).html(initialValue);
```

Comment:
With a `console.log(initialValue)` I'm now seeing what you mean about `initialValue` being updated with every input:

![0a8gjvqq1p](https://cloud.githubusercontent.com/assets/134455/16043082/fa4f7214-320b-11e6-83ae-af9d04329947.gif)


Here's me a) clicking the comment box, b) hitting delete, and c) clicking outside the comment box:

![hw9xrijihj](https://cloud.githubusercontent.com/assets/134455/16043012/9a3be894-320b-11e6-931f-73d179ff3975.gif)

What I'm missing is, what resets the value to the empty string when I click outside of the comment box?